packages/adminrouter/extra/src/common/util.lua

```diff
@@ -37,6 +37,31 @@ function util.get_file_content(path)
 end
 
 
+function util.verify_ip(ip)
+  -- Based on http://stackoverflow.com/a/16643628
+  -- Return True if ip is a valid IPv4 IP, False otherwise.
+  if type(ip) ~= "string" then return false end
+
+  -- check for format 1.11.111.111 for ipv4
+  local chunks = {ip:match("(%d+)%.(%d+)%.(%d+)%.(%d+)")}
```

Comment:
Probably silly but I think to be perfectly correct we want ^ in front and $ at the back to anchor the pattern. Otherwise `aaa.10.0.20.3` probably passes this test.